presto-tests/src/test/java/io/prestosql/execution/TestEventListener.java

```diff
@@ -260,14 +281,19 @@ private void assertFailedQuery(Session session, @Language("SQL") String sql, boo
         assertEquals(expectedFailure, failureInfo.getFailureMessage().orElse(null));
     }
 
-    private EventFilters buildEventFilters(boolean ignoreSplits)
+    private EventFilters buildEventFilters(String queryMarker, boolean ignoreSplits)
     {
+        Instant now = Instant.now();
         EventFilters.Builder builder = EventFilters.builder()
-                .setQueryCreatedFilter(event -> !event.getMetadata().getQuery().contains(IGNORE_EVENT_MARKER))
-                .setQueryCompletedFilter(event -> !event.getMetadata().getQuery().contains(IGNORE_EVENT_MARKER));
-        if (ignoreSplits) {
-            builder.setSplitCompletedFilter(event -> false);
-        }
+                .setQueryCreatedFilter(event -> {
+                    boolean queryOk = event.getMetadata().getQuery().contains(queryMarker);
+                    if (!queryOk) {
+                        ignoredQueries.add(event.getMetadata().getQueryId());
+                    }
+                    return queryOk;
+                })
+                .setQueryCompletedFilter(event -> !ignoredQueries.contains(event.getMetadata().getQueryId()))
+                .setSplitCompletedFilter(event -> !ignoreSplits && !ignoredQueries.contains(event.getQueryId()));
```

Comment:
split evens complicate things because you need ignoredQueries... but in fact, it seems no-one here is testing them. Am i missing something? we could just always ignore them and adjusted number of awaited events in `assertFailedQuery`. We would reiterate when we add actual tests for these events

Reply:
Oh - good point. I was not thinking about it. Maybe we do not care about those. But on the other hand we do validate if they are sent (to some extend) via event-count assertionts. Do you think it would be ok to just ignore those?
I think it would be nice to add a separate test to just test split events then. 

....

But it will open the problem with races again.